lib/tree.rb

```diff
@@ -1,38 +1,63 @@
 class NoApplesError < StandardError; end
 
 class AppleTree
-  attr_#fill_in :height, :age, :apples, :alive
+
+  attr_reader :height, :age, :apples, :alive
 
   def initialize
+    @age = 0
+    @height = 0
+    @apples = []
+    @alive = true
   end
 
   def age!
+    if alive
+      @age = @age + 1
+      @height = @height + 1
+      self.add_apples
+      if age >= 25
+        @alive = false
+      end
+    end
+    @age
```

Comment:
The return value of `!` methods should be irrelevant. They should only mutate the object.